### PR TITLE
AWS add SNS Topic Config support 

### LINKED ISSUE #5976: AWS config source pass supplementaryConfiguration data with resource
**Is your feature request related to a problem? Please describe.**
Any reason why we do not return `supplementaryConfiguration` along with `configuration` . This would be helpful since this seems to contain tag information on resources which would be useful for further value filters.

https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/query.py#L350

**Describe the solution you'd like**
We can either add it to the resource data by default , or can create an option on the `config` source that will then add it.

for example:

```
policies: 
  - name: sns-test
    resource: aws.sns
    source: config
    supplimentary-config: true
    query:  
      - clause:  "tags.key = 'tag-check'"
```

### PR DESCRIPTION
**Pull request by anovis**
sns topic is now a valid aws config resource 

closes #5976

### CODE REVIEW

File: tests/test_policy.py [hunk 1]
```diff
@@ -297,6 +297,8 @@ def test_config_resource_support(self):
                 "Missing config types \n %s" % ('\n'.join(missing)))
 
         bad_types = resource_config_types.difference(config_types)
+        # TODO: remove once botocore is updated https://github.com/boto/botocore/issues/2111
```
Comment (kapilt) [inline tests/test_policy.py:300]: its already present, the pinned version in custodian needs updating

Comment (anovis) [reply tests/test_policy.py:300]: o really. i tried with botocore version 1.17.26 and still got the same error as sns topic was missing. ill look again and see if i missed something. thanks

Comment (kapilt) [reply tests/test_policy.py:300]: 1.17.27 is released with the fix, i've got a separate pr to update the deps open.

Comment (kapilt) [reply tests/test_policy.py:300]: if you rebase on trunk, it should work, and then you can remove the whitelist entry in the config test_policy.py  coverage test

Comment (anovis) [reply tests/test_policy.py:300]: awesome thanks!

File: c7n/resources/sns.py [hunk 2]
```diff
@@ -41,6 +41,15 @@ def _augment(r):
             return list(w.map(_augment, resources))
 
 
+class ConfigSNS(ConfigSource):
+
+    def load_resource(self, item):
+        resource = super().load_resource(item)
```
Comment (kapilt) [inline c7n/resources/sns.py:47]: this needs a unit test for coverage

**Pull request merged**